src/zig.rs

```diff
@@ -715,8 +726,11 @@ impl Zig {
         // https://github.com/ziglang/zig/blob/0.9.1/src/Compilation.zig#L3408-L3445
         // https://github.com/ziglang/zig/blob/0.10.0/src/Compilation.zig#L4163-L4211
         // https://github.com/ziglang/zig/blob/0.10.1/src/Compilation.zig#L4240-L4288
-        if raw_target.contains("musl") {
+        if raw_target.contains("musl") || raw_target.contains("ohos") {
             args.push("-D_LIBCPP_HAS_MUSL_LIBC".to_owned());
+            // for musl or openharmony
+            // https://github.com/ziglang/zig/pull/16098
+            args.push("-D_LARGEFILE64_SOURCE".to_owned());
```

Comment:
I'm uncertain whether this is applicable to all MUSL outputs, but it's certainly required for OpenHarmony. Any thoughts on this?

Reply:
Ideally we should detect musl version (can be based on zig version I think) and only define this for musl 1.2.4+.